src/components/Modals/CreateSingleItemModal/ImportStep/ImportStep.tsx

```diff
@@ -44,58 +43,34 @@ export default class ImportStep extends React.PureComponent<Props, State> {
    *
    * @param file - The ZIP file.
    */
-  handleZippedModelFiles = async (file: File) => {
-    const zip: JSZip = await JSZip.loadAsync(file)
-    const fileNames: string[] = []
+  handleZippedModelFiles = async (file: File): Promise<ModelData> => {
+    const loadedFile = await loadFile(file.name, file)
+    const { wearable, content } = loadedFile
 
-    zip.forEach(fileName => {
-      if (!basename(fileName).startsWith('.')) {
-        fileNames.push(fileName)
-      }
-    })
-
-    // asset.json contains data to populate parts of the state
-    const assetJsonPath = fileNames.find(path => basename(path) === ASSET_MANIFEST)
     let assetJson: ItemAssetJson | undefined
-
-    if (assetJsonPath) {
-      const assetRaw = zip.file(assetJsonPath)
-      const content = await assetRaw.async('text')
-      assetJson = JSON.parse(content)
+    let modelPath: string | undefined
+
+    // wearable.json contains data to populate parts of the state
+    if (wearable) {
+      modelPath = getModelPath(wearable.data.representations)
+      assetJson = {
+        name: wearable.name,
+        description: wearable.description,
+        rarity: wearable.rarity,
+        category: wearable.data.category,
+        bodyShape: getBodyShapeType(wearable as Item)
+      }
+    } else {
+      modelPath = Object.keys(content).find(isModelPath)
     }
 
-    const modelPath = fileNames.find(isModelPath)
-
-    const files = await Promise.all(
-      fileNames
-        .map(fileName => zip.file(fileName))
-        .filter(file => !!file)
-        .map(async file => {
-          const blob = await file.async('blob')
-
-          if (blob.size > MAX_FILE_SIZE) {
-            throw new FileTooBigError()
-          }
-
-          return {
-            name: file.name,
-            blob
-          }
-        })
-    )
-
-    const contents = files.reduce<Record<string, Blob>>((contents, file) => {
-      contents[file.name] = file.blob
-      return contents
-    }, {})
-
     if (!modelPath) {
       throw new MissingModelFileError()
     }
 
-    const result = await this.processModel(modelPath, contents)
+    const result = await this.processModel(modelPath, content)
```

Comment:
The result of `processModel` will ignore one of the representations, I think we should change the code so we can return from this function all the data we can to use it later when building the item.

Reply:
Sorry a bit confused here. I think process model does not deal with representations?